tests/phpunit/api/v4/Entity/SubscriptionHistoryTest.php

```diff
@@ -62,4 +62,43 @@ public function testGet() {
     $this->assertLessThanOrEqual(time(), strtotime($historyRemoved->single()['date']));
   }
 
+  public function testGetPermissions() {
+    $this->createLoggedInUser();
+
+    $contact = $this->createTestRecord('Contact');
+    $group = $this->createTestRecord('Group');
+    $groupContact = $this->createTestRecord('GroupContact', [
+      'group_id' => $group['id'],
+      'contact_id' => $contact['id'],
+    ]);
+
+    \CRM_Core_Config::singleton()->userPermissionClass->permissions = [
+      'access CiviCRM',
+      'view all contacts',
+    ];
+
+    $historyAdded = SubscriptionHistory::get()
+      ->addSelect('*')
+      ->addWhere('group_id', '=', $group['id'])
+      ->addWhere('status', '=', 'Added')
+      ->addWhere('contact_id', '=', $contact['id'])
+      ->execute();
+    $this->assertCount(1, $historyAdded);
+
+    \CRM_Core_Config::singleton()->userPermissionClass->permissions = [];
+
+    try {
+      $historyAdded = SubscriptionHistory::get()
+        ->addSelect('*')
+        ->addWhere('group_id', '=', $group['id'])
+        ->addWhere('status', '=', 'Added')
+        ->addWhere('contact_id', '=', $contact['id'])
+        ->execute();
+      $this->assertCount(0, $historyAdded);
+    }
+    catch (UnauthorizedException $e) {
+    }
```

Comment:
I think the try/catch block should be removed so errors aren't invisible

Reply:
Thanks @colemanw. I've made that change.